src/test/java/org/alfresco/util/CachingDateFormatTest.java

```diff
@@ -0,0 +1,43 @@
+package org.alfresco.util;
+
+import org.junit.Test;
+
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Locale;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class CachingDateFormatTest
+{
+    private final LocalDateTime REFERENCE_DATE_TIME = LocalDateTime.of(2018, 4, 1, 10, 0); //2018-04-01 at 10:00am
+    
+    @Test
+    public void solrDatetimeFormat_DateNotUTC_shouldReturnISO8601DateString()
+    {
+        Instant shanghaiInstant = REFERENCE_DATE_TIME.atZone(ZoneId.of("Asia/Shanghai")).toInstant();
+        Date shanghaiDate = Date.from(shanghaiInstant);
+        SimpleDateFormat solrDatetimeFormat = CachingDateFormat.getSolrDatetimeFormat();
+        
+        String formattedDate = solrDatetimeFormat.format(shanghaiDate);
+        
+        assertThat(formattedDate,is("2018-04-01T02:00:00.000Z"));
+    }
+
+    @Test
+    public void solrDatetimeFormat_LocaleNotWestern_shouldReturnISO8601DateString()
+    {
+        Locale.setDefault(new Locale("hi", "IN"));
```

Comment:
Will this affect all subsequent tests?  Perhaps we should set it back after the test?  (Also, out of interest, is this Hindi? Maybe it's worth saying why you're picking this locale - e.g. it defaults to a different number system)

Reply:
Done, you are absolutely right!